### ISSUE TITLE
[Bug]: .Net Maui -> Notification Permission Prompt is not appearing.

### ISSUE THREAD

**Issue by palanikumar32** [OneSignal/OneSignal-DotNet-SDK#63]
### What happened?

Notification Permission Prompt not appearing if one signal is not initialized in constructor of App.xaml.cs or MainPage.xaml.cs
We want to subscribe one signal only after the user is logged in. So we moved the below line to login button click
`OneSignal.Initialize("xxxxxxxxxxx");`
after that Notification Permission Prompt is not appearing. We tried both
`OneSignal.User.PushSubscription.OptIn();`
and 
`OneSignal.InAppMessages.AddTrigger("showPrompt", "true");`

### Steps to reproduce?

```Markdown
Remove the below line in App start [public App(){ OneSignal.Initialize("xxxxxxxxxxx"); }]
OneSignal.Initialize("xxxxxxxxxxx");
```

### What did you expect to happen?

Notification Permission Prompt is not appearing.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by emawby** [comment]
@palanikumar32 Could you try `OneSignal.Notifications.RequestPermissionAsync()`

**Comment by palanikumar32** [comment]
> @palanikumar32 Could you try `OneSignal.Notifications.RequestPermissionAsync()`

Thanks for the reply, already tried no luck.

**Comment by emawby** [comment]
@palanikumar32 Could you try enabling verbose logging and reporting what you see in the logs? Do any other OneSignal methods work? Also is the behavior the same for both iOS and Android?

**Comment by palanikumar32** [comment]
> @palanikumar32 Could you try enabling verbose logging and reporting what you see in the logs? Do any other OneSignal methods work? Also is the behavior the same for both iOS and Android?

Here i attached the logs, unfortunately right now I couldn't test the same behavior in IOS.
Problem is our App will get onesignal id only after login, so we want to trigger OneSignal.Initialize and permission dialog only on login success.
[log-initialize-button-click.txt](https://github.com/OneSignal/OneSignal-DotNet-SDK/files/12503416/log-initialize-button-click.txt)
[log-subscribe-button-click.txt](https://github.com/OneSignal/OneSignal-DotNet-SDK/files/12503417/log-subscribe-button-click.txt)
[OneSignalMaui-testapp.zip](https://github.com/OneSignal/OneSignal-DotNet-SDK/files/12503418/OneSignalMaui-testapp.zip)

**Comment by emawby** [comment]
@palanikumar32 I apologize for the late response. I am not able to reproduce this issue on Android or iOS. We are doing a release soon that updates the native iOS and Android SDK dependencies, uses .net7 tools instead of legacy Xamarin build tools, and fixes other various bugs. When that release is available I will ping here again and we can try to reproduce. I believe that this issue may be resolved by that release.

**Comment by jennantilla** [comment]
Hello everyone, apologies that this issue has gone stale. Please upgrade to the latest version of the SDK and open a new issue if you are still having trouble. Thanks!
